app/src/Page/Menu.elm

```diff
@@ -12,76 +17,291 @@ import Bulma
 import Route
 import Json.Decode as Decode
 import Json.Encode as Encode
+import Autocomplete
+import Date exposing (Date)
+import Date.Format
+
+type alias Retro =
+    { id : String
+    , name : String
+    , createdAt : Date
+    , participants : List String
+    }
 
 type alias Model =
-    { retroList : Maybe (List String)
+    { retroList : List Retro
     , retroName : String
+    , possibleParticipants : List String
+    , participants : List String
+    , participant : String
+    , autocompleteState : Autocomplete.State
+    , currentChoice : Maybe Retro
     }
 
-init : (Model, Cmd Msg)
-init =
-    { retroList = Nothing
+updateConfig : Autocomplete.UpdateConfig Msg String
+updateConfig =
+    Autocomplete.updateConfig
+        { toId = identity
+        , onKeyDown =
+              \code maybeId ->
+                  if code == 13 then
+                      Maybe.map SelectParticipant maybeId
+                  else
+                      Nothing
+        , onTooLow = Nothing
+        , onTooHigh = Nothing
+        , onMouseEnter = \_ -> Nothing
+        , onMouseLeave = \_ -> Nothing
+        , onMouseClick = \id -> Just <| SelectParticipant id
+        , separateSelections = False
+        }
+
+empty : Model
+empty =
+    { retroList = []
     , retroName = ""
-    } ! [ getRetros ]
+    , possibleParticipants = []
+    , participants = []
+    , participant = ""
+    , autocompleteState = Autocomplete.empty
+    , currentChoice = Nothing
+    }
 
-type Msg = GotRetros (Result Http.Error (List String))
-         | CreateRetro
-         | SetRetroName String
+mount : Sock.Sender Msg -> Cmd Msg
+mount sender =
+    Sock.menu sender
 
-getRetros : Cmd Msg
-getRetros =
-     Http.get "/retros" (Decode.list Decode.string)
-         |> Http.send GotRetros
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.map SetAutoState Autocomplete.subscription
```

Comment:
the only thing this does is activate a spinner I don't care about, so maybe remove it